client/components/domains/registrant-extra-info/ca-form.jsx

```diff
@@ -117,12 +117,6 @@ export class RegistrantExtraInfoCaForm extends React.PureComponent {
 		const { value, checked, type, id } = event.target;
 		const newContactDetails = {};
 
-		if ( id === 'legal-type' ) {
-			this.props.updateRequiredDomainFields?.( {
```

Comment:
If you select "Canadian Corporation" in checkout with a `.ca` domain in the cart, then validation still displays an error correctly:

<img width="519" alt="Screen Shot 2021-09-03 at 5 19 46 PM" src="https://user-images.githubusercontent.com/2036909/132066340-183c8d78-15d2-4741-9ac4-9da20b3141fe.png">

Reply:
On the calypso.live build and loading the branch directly, the Organization box does not seem to appear when I have selected Canadian Citizen, save the contact info, and then edit and try selecting Canadian Corporation.

If I hit continue at this point, I receive the error `We could not validate your contact information. Please review and update all the highlighted fields.` but no highlighted fields are shown. I presume the erroring field is the non-visible Organization field.

![Screen Shot 2021-09-07 at 11 33 10 AM](https://user-images.githubusercontent.com/8002138/132394057-429e073a-23e5-4967-82e5-0a255eefbfee.png)

Reply:
Wow, you're right! Without this PR, if I select "Canadian Corporation", the _Organization_ field automatically becomes visible!

Doing some digging, it appears to come from this code where the `isRequired` value is explicitly checked for the _Organization_ field:

https://github.com/Automattic/wp-calypso/blob/3194533e2eb93f3fa3b853090538c957e6c24f5b/client/components/domains/contact-details-form-fields/managed-contact-details-form-fields.jsx#L308-L309

I don't actually see that mechanism used anywhere else, so I assume this is a special case, although I could see this being an issue in future forms if we remove the option to force otherwise hidden fields to be displayed.

I have four ideas:

1. Leave the `isRequired` machinery in place, but only for the purpose of forcing fields to be visible. This seems like a waste of resources considering the single use case and the hundreds of lines of code that are needed by required fields.
2. Add a new state mechanism to force fields to be visible. This is similar to option 1 but could be considerably smaller.
3. Force the _Organization_ field to be visible by explicitly examining the value of the field that makes it required. This could be very small and while it doesn't provide an abstraction, option 2 could always be added later if this pattern was repeated frequently.
4. Never force a field to be visible, but alter the validation error message to be specific about what fields are invalid, replacing `Please review and update all the highlighted fields.` with something like `There is a problem with the "Organization" field.` (Maybe it only mentions one of the failing fields?) This is less convenient than the other options, but would probably get the point across. This could be done in parallel to one of the other changes and might also help prevent confusion in other cases as well.

Reply:
I pushed option 3. The Org field should always be visible now if required.

Reply:
i re-tested this, and it seems to now work correctly.